source/dotnet/Library/AdaptiveCards/Rendering/AdaptiveVisitor.cs

```diff
@@ -1,56 +1,67 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+using AdaptiveCards;
+using System.Collections.Generic;
+
 namespace AdaptiveCards.Rendering
 {
     /// <summary>
     ///     Utility class which allows you to visit every node
     /// </summary>
     public class AdaptiveVisitor
     {
+        public HashSet<AdaptiveTypedElement> Elements { get; set; } = new HashSet<AdaptiveTypedElement>();
+
         public virtual void Visit(AdaptiveElement adaptiveElement)
         {
             if (adaptiveElement is AdaptiveImage image)
                 Visit(image);
 
-            if (adaptiveElement is AdaptiveTextBlock textBlock)
+            else if (adaptiveElement is AdaptiveTextBlock textBlock)
```

Comment:
appreciate the optimization here -- I wonder if we can go further with something like:

```c#
switch (adaptiveElement)
{
case AdaptiveTextBlock textBlock:
    Visit(textBlock);
    break;
case AdaptiveContainer container:
    Visit(container);
    break;
/// ...
};
```

Reply:
yeah, I considered this but it doesn't save anything on spacing or typing so ...?shrug?

Reply:
I was thinking that it might be quicker, but ultimately probably not terribly important :)